Sources/Core/TypedNode.swift

```diff
@@ -135,10 +135,17 @@ extension TypedNode where ID: DeclID {
   }
 
   /// The type of the declared entity.
+  /// In most cases, the type is valid, so we extract the value from the optional by default.
   public var type: AnyType {
     program.declTypes[id]!
```

Comment:
If we do as below, we don't need forced unwrapping anymore and we can represent failures to type check an expression with the error type.

```suggestion
    program.declTypes[id] ?? .error
```

Reply:
Makes a lot of sense. Will fix.